src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -305,9 +305,14 @@ private void RegisterLocalServices()
             this.localServices.Add(typeof(IRuleSetSerializer), new Lazy<ILocalService>(() => new RuleSetSerializer(this)));
             this.localServices.Add(typeof(IConfigurationProvider), new Lazy<ILocalService>(() =>
             {
+                var solution = this.GetService<SVsSolution, IVsSolution>();
+                var store = new SecretStore(SolutionBindingSerializer.StoreNamespace);
                 var legacySerializer = new SolutionBindingSerializer(this);
-                return new ConfigurationProvider(legacySerializer, InMemoryConfigurationProvider.Instance);
-
+                // The SCC wrapper maintains a queue of file writes. For the new provider we want to the file write to be
+                // executed immediately, so we want our own SCC wrapper rather than sharing the one used by the legacy mode.
+                var sccFileSystem = new SourceControlledFileSystem(this);
+                var newConfigSerializer = new ConfigurationSerializer(solution, sccFileSystem, store, Logger);
+                return new ConfigurationProvider(legacySerializer, newConfigSerializer);
```

Comment:
This code will change again shortly once the new binding method is implemented (we'll be writing multiple files during binding again, so it should be ok to share the SCC wrapper again).